src/main/java/org/gridsuite/explore/server/services/DirectoryService.java

```diff
@@ -87,6 +87,25 @@ public ElementAttributes createElementWithNewName(ElementAttributes elementAttri
                 .getBody();
     }
 
+    public ElementAttributes duplicateElement(UUID elementUuid, UUID newElementUuid, UUID targetDirectoryId, String userId) {
+        UriComponentsBuilder uri = UriComponentsBuilder
+                .fromPath(ELEMENTS_SERVER_ROOT_PATH)
+                .queryParam("duplicateFrom", elementUuid)
+                .queryParam("newElementUuid", newElementUuid);
+        if (targetDirectoryId != null) {
+            uri.queryParam("targetDirectoryId", targetDirectoryId);
+        }
+        String path = uri.buildAndExpand(elementUuid)
```

Comment:
you want to buildAndExpand a property which is absent in the path again

Reply:
done